logpyle/__init__.py

```diff
@@ -707,6 +711,10 @@ class WatchInfo(Record):
 
             self.watches.append(watch_info)
 
+    def set_watch_interval(self, interval: float) -> None:
+        """Set the interval (in seconds) between the time watches are printed."""
+        self.watch_interval = interval
```

Comment:
Make this a constructor parameter instead? Is there a use case for setting this as the code runs?

Reply:
> Make this a constructor parameter instead? 

We could also add it as a constructor parameter on top of the method.

> Is there a use case for setting this as the code runs?

I could think of a case where you'd want to check a watch frequently (say, after starting or restarting an app), and check less frequently after some time has passed.

Reply:
> I could think of a case where you'd want to check a watch frequently at some point (say, after starting or restarting an app), and check less frequently after some time has passed.

I like this quite a bit.  You can watch every step for the first 10, then tune it down to every 100, or just every I/O step or whatever.  Cool.

Reply:
I also imagine that changing the log frequency at runtime could screw with lazy?

Reply:
Nope, lazy doesn't care one bit. This all happens outside of it.

Reply:
OK, fine by me... but do also add a constructor argument.

Reply:
Added